src/components/AnalysisPage/3dModelViewer/setupThreeScene.js

```diff
@@ -46,23 +47,28 @@ const setupThreeScene = (canvas, objectURL) => {
 
   const disposables = [];
 
-  const loader = new STLLoader();
+  const loader = fileType === "stl" ? new STLLoader() : new PLYLoader();
```

Comment:
This could be made more explicit. so like actually checking for a fileType of ply when choosing the PLYLoader rather than just defaulting to it if its not STL. And then if the fileType is neither stl or ply it could throw an error. This way if we later modify something and the fileType becomes weird it will be a lot easier to understand the issue. If the fileType becomes wrong then we'll see an error rather than it just defaulting to ply viewer

Reply:
@rknizzle 

Yes for sure. 

The only issue is when the file type isn't supported, the whole application breaks. 

I'm looking into wrapping the modelViewer component in an error-boundary component to display some sort of UI instead of everything falling apart.

But we probably don't even need that error-boundary component's UI because we would never deploy code that sends a bad file type to the 3dModelViewer... We'll just see the error and fix it.

Right?

Reply:
Yea I think you're right here. It probably wont happen that an unsupported file type would be passed to here. It'll just be something that we need to fix if it ever does happen